plugin/forward/persistent.go

```diff
@@ -157,7 +157,18 @@ func (t *transport) Yield(c *dns.Conn) {
 }
 
 // Stop stops the transport's connection manager.
-func (t *transport) Stop() { t.stop <- true }
+func (t *transport) Stop() {
+	t.stop <- true
+
+	// at this point nothing should write to channels anymore
+	// closing channels to make tests panic on write
```

Comment:
I find this `close`-list somewhat ugly, is this only needed for the test? If so can we just do it in the tests only?

Reply:
Ok, I will remove this.
The idea was is that panic on writing in closed channel is easier to detect and fix than hanging when nobody reads the channel